README.md

```diff
@@ -67,6 +67,199 @@ Or install the plugin manually:
+        # add breakpoints like:
+        import pydevd_pycharm
+        pydevd_pycharm.settrace('localhost', port=53100, stdoutToServer=True, stderrToServer=True)
+    ```
+    avoid raising exceptions in `ORStools/gui/ORStoolsDialog.py` to not crash QGIS every time one is raised
+    ```python
+    # below other imports
+    from ..ORStoolsPlugin import DEBUG
+   
+    # surround raise with if block around run_gui_control()
+    if not DEBUG:
+        raise
+    ```
+   commit changes
+   ```shell
+    git add . && git commit -m "Debug commit"
+   ```
+**Important**: When using the remote debugger of PyCharm you have to **disable the First Aid plugin**, as it interferes
+with the remote debugger.
+
+#### Workflow
+
+To debug you now only need to `cherry-pick` the debug commit to the branch you are working on and place any changes
+on top.
+```shell
+# this will cherry pick the last commit of the debug branch
+git cherry-pick debug
+```
+
+Make sure the local `debug` branch is up to date with the `main` branch by rebasing regularly 
+```shell
+# you will be on debug branch afterwards
+git rebase main debug
+```
+
+Before starting QGIS, you need to run the "QGIS debug" configuration you created.  
+Afterwards you can open QGIS and press the plugin reloader button (configured to reload ORStools).
+It should break at the breakpoint introduced in the debug commit.
+
+In general, you can now use normal breakpoints of the IDE with [left click in the gutter](https://www.jetbrains.com/help/pycharm/using-breakpoints.html#set-line-breakpoint)
+(or ctrl/cmd + F8).
+
+If you are debugging the processing algorithms, which run in another thread, you will have to add another manual
+breakpoint in e.g. `ORStools/proc/isochrones_layer_proc.py` by typing `br` (or
+whatever you configured in your live template), pressing enter and **reload the plugin** in QGIS.
+
+In short: Use IDE breakpoints if they work, if not use manual and IDE breakpoints afterwards.
+
+Once you finalized your changes, remove the manual breakpoints again and drop the debug commit.
+
+You can do this with one of the following
+- pressing `alt/option + 9` and right-click the debug commit on your branch and choose `Drop Commit`
+- `git stash && git reset --hard HEAD^ && git stash pop`
+- commit your changes and `git rebase -i HEAD^^`, prepend the debug commit with a `d` and save
+
+### Interface development
+
+For designing the Dialog the Qt designer shipping with qgis is used. It has relevant classes such as
+[QgsMapLayerComboBox](https://qgis.org/pyqgis/3.2/gui/Map/QgsMapLayerComboBox.html) already imported properly.
+
+#### Mac
+- use `/Applications/QGIS-LTR.app/Contents/MacOS/bin/designer` instead of
+  `/Applications/QGIS-LTR.app/Contents/MacOS/Designer.app` (which is not working, and trying to get other Qt Designer
+  or Qt Creator installations to use the correct QGIS classes is a nightmare that cost me 3 days of my life)
```

Comment:
While this is a fun story to tell - it shouldn't be mentioned here, at least not like this ;)

Reply:
rephrased